identity/handler.go

```diff
@@ -183,11 +179,73 @@ type listIdentitiesParameters struct {
 	crdbx.ConsistencyRequestParameters
 }
 
+func parseListIdentitiesParameters(r *http.Request) (params ListIdentityParameters, err error) {
+	query := r.URL.Query()
+	var requestedFilters int
+
+	params.Expand = ExpandDefault
+
+	if ids := query["ids"]; len(ids) > 0 {
+		requestedFilters++
+		for _, v := range ids {
+			id, err := uuid.FromString(v)
+			if err != nil {
+				return params, errors.WithStack(herodot.ErrBadRequest.WithReasonf("Invalid UUID value `%s` for parameter `ids`.", v))
+			}
+			params.IdsFilter = append(params.IdsFilter, id)
+		}
+	}
+	if len(params.IdsFilter) > 500 {
+		return params, errors.WithStack(herodot.ErrBadRequest.WithReason("The number of ids to filter must not exceed 500."))
+	}
+
+	if orgID := query.Get("organization_id"); orgID != "" {
+		requestedFilters++
+		params.OrganizationID, err = uuid.FromString(orgID)
+		if err != nil {
+			return params, errors.WithStack(herodot.ErrBadRequest.WithReasonf("Invalid UUID value `%s` for parameter `organization_id`.", orgID))
+		}
+	}
+
+	if identifier := query.Get("credentials_identifier"); identifier != "" {
+		requestedFilters++
+		params.Expand = ExpandEverything
+		params.CredentialsIdentifier = identifier
+	}
+
+	if identifier := query.Get("credentials_identifier_similar"); identifier != "" {
+		requestedFilters++
+		params.Expand = ExpandEverything
+		params.CredentialsIdentifierSimilar = identifier
+	}
+
+	for _, v := range query["include_credential"] {
+		params.Expand = ExpandEverything
+		tc, ok := ParseCredentialsType(v)
+		if !ok {
+			return params, errors.WithStack(herodot.ErrBadRequest.WithReasonf("Invalid value `%s` for parameter `include_credential`.", v))
+		}
+		params.DeclassifyCredentials = append(params.DeclassifyCredentials, tc)
+	}
+
+	if requestedFilters > 1 {
+		return params, errors.WithStack(herodot.ErrBadRequest.WithReason("You cannot combine multiple filters in this API"))
+	}
```

Comment:
We could consider to reduce the impact of this breaking change by just ignoring additional filters. If `ids=a&ids=b&organization_id=x` previously returned only `a`, it could now return `a,b` ignoring the second filter. The result should always be a super-set of the previous version, therefore significantly limiting the impact.
On the other side, someone might rely on the behavior and this could in the worst case lead to security issues if the assumption is that all returned identities are part of the given organization.
WDYT?

Reply:
I've checked our logs and could not find anyone combining filters, so I think we're good.

Changing semantics silently is a recipe for disaster.